qtconsole/console_widget.py

```diff
@@ -207,8 +207,7 @@ def __init__(self, parent=None, **kw):
         parent : QWidget, optional [default None]
             The parent for this widget.
         """
-        QtGui.QWidget.__init__(self, parent)
-        LoggingConfigurable.__init__(self, **kw)
+        super(ConsoleWidget, self).__init__(**kw)
```

Comment:
Well i haven't really checked, but i assume it is violating the user's expectations that passing in a parent would set the parent widget. Quoting from the Qt docs for `QWidget::QWidget ( QWidget * parent = 0, Qt::WindowFlags f = 0 )`
> If parent is 0, the new widget becomes a window. If parent is another widget, this widget becomes a child window inside parent. The new widget is deleted when its parent is deleted

The following code should show a `ConsoleWidget` inside the `QWidget`, which fails if the parent is not set.
```
w = QtGui.QWidget()
cw = ConsoleWidget(w)
w.show()  # Should show a ConsoleWidget within the window
```

You can simply set the parent in here via a `self.setParent(parent)` call after the `__init__()` call. It actually is not a problem in practice because we all use the `ConsoleWidget` within a layout, which automatically manages the parenting of its child widgets.